### PR TITLE
[devconsole] Keystore in devconsole, support stored wallets

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Devconsole.ts now uses KeyStore to store wallets persistently.
See https://github.com/trustwallet/wallet-core/blob/ar/devconsole-keystore/samples/typescript/devconsole.ts/samplescripts/wallets.sampleinput.txt for demo script

## How to test

```
cd samples/typescript/devconsole.ts
npm install
npm run build

echo "await walletCreate(256, 'TEST'); \n walletDump(); \n .exit" | npm run start

echo "await walletDump() \n .exit" | npm run start
```

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [ ] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [ ] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

File: samples/typescript/devconsole.ts/src/index.ts [hunk 1]
```diff
@@ -26,61 +26,196 @@ function enumerateNamespaces(topLevelNamespace: any) {
...
+            /*
+            const hdWallet = WC.HDWallet.createWithMnemonic(mnemonic, this.HDWalletPassord);
+            const storedKey = WC.StoredKey.importHDWallet(hdWallet.mnemonic(), name, Buffer.from(this.StoreFixedPassword), this.DefaultCoins[0]);
+            this.DefaultCoins.forEach((coin) => {
+                storedKey.accountForCoin(coin, hdWallet);
+            });
+            let wallet = this.keystore.mapWallet(storedKey);
+            await this.keystore.importWallet(wallet);
+            this.wallet = new StoredKeyWallet(storedKey);
+            */
```
Comment (hewigovens) [inline samples/typescript/devconsole.ts/src/index.ts:142]:
```suggestion
```

File: samples/typescript/devconsole.ts/src/index.ts [hunk 2]
```diff
@@ -26,61 +26,196 @@ function enumerateNamespaces(topLevelNamespace: any) {
...
+        StoreFixedPassword = "devconsole.ts";
+        HDWalletPassord = '';
+        DefaultCoins: any[] = [];
+        keystore: any;
+        // wallets available in storage
+        storeWallets: any[] = [];
+        
+        init(): void {
+            this.DefaultCoins = [WC.CoinType.bitcoin, WC.CoinType.ethereum, WC.CoinType.binance];
+            fs.mkdir(this.StoreFolderPath, () => {});
```
Comment (miloserdow) [inline samples/typescript/devconsole.ts/src/index.ts:69]: let's handle cases when this fails to work

File: samples/typescript/devconsole.ts/src/index.ts (hunk 1 above)
Comment (catenocrypt) [reply samples/typescript/devconsole.ts/src/index.ts]: Removed commented out block, added comment about other possibility

File: samples/typescript/devconsole.ts/src/index.ts (hunk 2 above)
Comment (catenocrypt) [reply samples/typescript/devconsole.ts/src/index.ts]: Good point, added error handling for Filesystem problems.

**Review by hewigovens — APPROVED** [review]

**Review by miloserdow — APPROVED** [review]

**Pull request merged**